pkg/controllers/apps/feedinventory/feedinventory.go

```diff
@@ -0,0 +1,366 @@
+		c.workqueue.Forget(obj)
+		klog.Infof("[FeedInventory] successfully synced Subscription %q", key)
+		return nil
+	}(obj)
+
+	if err != nil {
+		utilruntime.HandleError(err)
+		return true
+	}
+
+	return true
+}
+
+// syncHandler compares the actual state with the desired, and attempts to
+// converge the two. It then updates the Status block of the Subscription resource
+// with the current status of the resource.
+func (c *Controller) syncHandler(key string) error {
+	// If an error occurs during handling, we'll requeue the item so we can
+	// attempt processing again later. This could have been caused by a
+	// temporary network failure, or any other transient reason.
+
+	// Convert the namespace/name string into a distinct namespace and name
+	ns, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("invalid resource key: %s", key))
+		return nil
+	}
+
+	klog.V(4).Infof("[FeedInventory] start processing Subscription %q", key)
+	// Get the Subscription resource with this name
+	sub, err := c.subLister.Subscriptions(ns).Get(name)
+	// The Subscription resource may no longer exist, in which case we stop processing.
+	if errors.IsNotFound(err) {
+		klog.V(2).Infof("Subscription %q has been deleted", key)
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	if sub.DeletionTimestamp != nil {
+		return nil
+	}
+
+	if sub.Spec.SchedulingStrategy != appsapi.DividingSchedulingStrategyType {
+		return nil
+	}
+
+	sub.Kind = subKind.Kind
+	sub.APIVersion = subKind.GroupVersion().String()
+
+	finv := &appsapi.FeedInventory{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      sub.Name,
+			Namespace: sub.Namespace,
+			Labels: map[string]string{
+				known.ObjectCreatedByLabel: known.ClusternetHubName,
+				// add subscription info
+				known.ConfigSubscriptionNameLabel:      sub.Name,
+				known.ConfigSubscriptionNamespaceLabel: sub.Namespace,
+				known.ConfigSubscriptionUIDLabel:       string(sub.UID),
+			},
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion:         sub.APIVersion,
+					Kind:               sub.Kind,
+					Name:               sub.Name,
+					UID:                sub.UID,
+					Controller:         utilpointer.BoolPtr(true),
+					BlockOwnerDeletion: utilpointer.BoolPtr(true),
+				},
+			},
+		},
+		Spec: appsapi.FeedInventorySpec{
+			Feeds: make([]appsapi.FeedOrder, len(sub.Spec.Feeds)),
+		},
+	}
+
+	wg := sync.WaitGroup{}
+	wg.Add(len(sub.Spec.Feeds))
+	errCh := make(chan error, len(sub.Spec.Feeds))
+	for idx, feed := range sub.Spec.Feeds {
+		go func(idx int, feed appsapi.Feed) {
+			defer wg.Done()
+
+			manifests, err2 := utils.ListManifestsBySelector(c.reservedNamespace, c.manifestLister, feed)
+			if err2 != nil {
+				errCh <- err2
+				return
+			}
+			if manifests == nil {
+				errCh <- apierrors.NewNotFound(schema.GroupResource{Resource: feed.Kind}, feed.Name)
+				return
+			}
+
+			gvk, err2 := getGroupVersionKind(manifests[0].Template.Raw)
```

Comment:
Does the manifest always have only one element?

Reply:
Yes.